### ISSUE TITLE
Improve the cost estimation to run a Calibrator

### ISSUE THREAD

**Issue by andreamari** [unitaryfund/mitiq#1715]
## Issue Description

The Calibrator class as a [get_cost()](https://github.com/unitaryfund/mitiq/blob/ac53f1dd700507a92fbcb0fdbe6757fccefd68a0/mitiq/calibration/calibrator.py#L88) method to estimate the number of circuit executions that are necessary to get all the required data. 

At the moment, the estimate is not correct since it doesn't take into account that each Strategy requires the execution of multiple circuits for a single error-mitigated expectation value (e.g. ZNE requires k circuits where k is the number of noise scale factors, PEC requires `num_samples` circuits, etc.. ).

This issue is about getting a more faithful estimate of the actual cost.

## Proposed Solution

Maybe each Strategy object could have a `num_circuits_required()` method, such that one can simply scale the result in [get_cost()](https://github.com/unitaryfund/mitiq/blob/ac53f1dd700507a92fbcb0fdbe6757fccefd68a0/mitiq/calibration/calibrator.py#L88)
by that number. 

Other ideas are welcome.

**Comment by Gopal-Dahale** [comment]
Hi @andreamari, the current model multiplies the number of circuits by the number of strategies. However, this model does not consider other factors that can affect the cost of calibration, such as the number of samples required for each error-mitigated expectation value, the number of noise scale factors etc. A more sophisticated model could consider these factors and provide a more accurate estimate of the cost.

As you suggested we can have that `num_circuits_required` method in each of the strategy objects or we can infer it from the existing attributes (`technique`, `scale_noise` and `factory`).

**Comment by andreamari** [comment]
Yes, indeed this issue asks for a better counting of the number of circuits, such that the `get_cost()` method would give a more accurate information.

**Comment by Gopal-Dahale** [comment]
I would like to contribute to this issue. May you assign it to me?

**Comment by andreamari** [comment]
This is an issue selected for [unitaryhack.dev](unitaryhack.dev). As a temporary rule for this event, we decided no to assign the issues to contributors until the associated PRs are approved and merged. This is to avoid the risk of stale assigments and stale PRs.

You are welcome to open a PR without an assignment. In that case we suggest [linking](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) your PR to this issue. If it will be successfully merged, you'll get the unitaryhack bounty :1st_place_medal:

**Comment by pandaconstantin** [comment]
Is it possible to grab this issue ?

**Comment by Misty-W** [comment]
> Is it possible to grab this issue ?

@pandaconstantin it's still possible, but there is a PR already open to solve this issue, and it will likely be approved soon. There are still plenty of open bouties for unitaryHACK though.

**Comment by YuNariai** [comment]
Thank you Nate!
